src/library_webgpu.js

```diff
@@ -91,6 +91,7 @@
     // Must be in sync with webgpu.h.
     COPY_STRIDE_UNDEFINED: 0xFFFFFFFF,
     LIMIT_U32_UNDEFINED: 0xFFFFFFFF,
+    WHOLE_MAP_SIZE: 0xFFFFFFFF, // 32bits max
```

Comment:
The implementation of wgpuBufferMapAsync needs to detect when size==WHOLE_MAP_SIZE and convert it to `undefined`.

Reply:
I see. Sine this flag is different from dawn_native one and it will ends in https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/webgpu/gpu_buffer.cc;l=182;drc=0af216b2753abe71a0f250a204f24fb4b1b0cb85;bpv=1;bpt=1 to get replaced.

But I tried adding the code snippet 
```javascript
if (size === {{{ gpu.WHOLE_MAP_SIZE }}}) {
  console.log('\n\n\n' + size + '\n\n' + {{{ gpu.WHOLE_MAP_SIZE }}} + '\n\n');
  size = undefined;
}
```
at `wgpuBufferMapAsync` in `library_webgpu.js`.

If I call mapAsync with `size == wgpu::kWholeMapSize`, by doing some printing the `wgpu::kWholeMapSize` is interpreted correctly as `0xffffffff`. But `size` passed to `wgpuBufferMapAsync` becomes -1, which I think is resulted from
```cpp
#define WGPU_WHOLE_MAP_SIZE SIZE_MAX
```
in the `webgpu.h`. 

I think the solution would be somehow modify dawn generator to put different value tag for emscripten-bits gen?
I think I've mostly figured this out already when writing this down...